app/controllers/sessions_controller.rb

```diff
@@ -5,10 +5,51 @@
 #   the COPYRIGHT file.
 
 class SessionsController < Devise::SessionsController
+  before_action :authenticate_with_2fa, only: [:create]
```

Comment:
Rails/LexicallyScopedActionFilter: `create` is not explicitly defined on the class.

Reply:
This one is tricky, `#create` comes from devise but I need to run the before action if a user has 2fa enabled